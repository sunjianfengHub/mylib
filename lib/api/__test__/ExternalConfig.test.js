jest.autoMockOff();
jest.mock('fs');
jest.mock('../../demon.test.config.js');

let externalConfig
beforeEach(() => {
	externalConfig = require('../ExternalConfig');
})
afterEach(() => {
	externalConfig = null;
})

describe('externalConfig Test', () => {
	test('default externalConfig is Falsy', () => {
		expect(externalConfig.externalConfig).toBeFalsy();
	})
	test('getExternalConfig without filename should return falsy', () => {
		expect(externalConfig.getExternalConfig()).toMatchSnapshot();
	})
	test('get none exist file should have empty object', () => {
		expect(externalConfig.getExternalConfig('none.js')).toMatchSnapshot();
	})
	test('get external config should have property', () => {
		// console.log(externalConfig.getExternalConfig('demon.test.config.js'));
		expect(externalConfig.getExternalConfig('demon.test.config.js')).toHaveProperty('server');
	})
	test('get external config without search file if got file', () => {
		let ApiTool = require('../ApiTool.js');
		const getlibRootSpy = jest.spyOn(ApiTool, 'getlibRootPath');
		let ret = externalConfig.getExternalConfig('demon.test.config.js');
		expect(getlibRootSpy).not.toHaveBeenCalled();
		getlibRootSpy.mockReset();
		getlibRootSpy.mockRestore();
	})
})