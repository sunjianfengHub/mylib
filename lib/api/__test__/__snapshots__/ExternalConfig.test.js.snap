// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`externalConfig Test get none exist file should have empty object 1`] = `Object {}`;

exports[`externalConfig Test getExternalConfig without filename should return falsy 1`] = `Object {}`;
